TimePlatformer/Assets/Scripts/EnemyHorizontalMovementScript.cs

```diff
@@ -0,0 +1,45 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class EnemyHorizontalMovementScript : MonoBehaviour {
+
+    [Range(-1, 1)] public int direction;
+
+    public float walkSpeed;
+
+    public float maxRange;
+
+    private Vector2 startPosition;
+
+    private Vector2 walkingVector;
+
+	// Use this for initialization
+	void Start () {
+        startPosition = transform.position;
+	}
+	
+	// Update is called once per frame
+	void Update () {
+        Walk();
```

Comment:
Why is this a separate function? It seems like the contents of Walk could just go into here.